pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -124,4 +124,14 @@
      * @return the key of the message
      */
     String getKey();
+
+    /**
+     * Get the topic name of this message.
+     * This is mainly for TopicsConsumerImpl to identify a message belongs to which topic.
+     *
+     * @return the topic name
+     */
+    default String getTopicName() {
```

Comment:
We shouldn't need to use default here, we can always return null in the implementation if needed.

Reply:
thanks. will change it.